packages/components/src/navigation/item/index.js

```diff
@@ -12,22 +12,32 @@ import { Icon, chevronRight } from '@wordpress/icons';
 /**
  * Internal dependencies
  */
-import Button from '../button';
-import { useNavigationContext } from './context';
-import { ItemBadgeUI, ItemTitleUI, ItemUI } from './styles/navigation-styles';
+import Button from '../../button';
+import { useNavigationContext } from '../context';
+import { ItemBadgeUI, ItemTitleUI, ItemUI } from '../styles/navigation-styles';
+import { useNavigationTreeItem } from './use-navigation-tree-item';
+import { useNavigationMenuContext } from '../menu/context';
 
-export default function NavigationItem( {
-	badge,
-	children,
-	className,
-	href,
-	item,
-	navigateToMenu,
-	onClick = noop,
-	title,
-	...props
-} ) {
+export default function NavigationItem( props ) {
+	useNavigationTreeItem( props );
```

Comment:
Mind moving this after the props destructuring, in order to keep all hooks together?

Reply:
Done